src/scripting/scriptingService.ts

```diff
@@ -55,11 +55,27 @@ export class ScriptingService {
         return scriptingObject;
     }
 
-    public async scriptSelect(node: TreeNodeInfo, uri: string): Promise<string> {
+    /**
+     * Helper to create scripting params
+     */
+    public createScriptingParams(node: TreeNodeInfo, uri: string, operation: ScriptOperation): ScriptingParams {
         const scriptingObject = this.getObjectFromNode(node);
         let serverInfo = this._connectionManager.getServerInfo(node.connectionCredentials);
+        let scriptCreateDropOption: string;
+        switch (operation) {
+            case (ScriptOperation.Select):
+                scriptCreateDropOption = 'ScriptSelect';
+                break;
+            case (ScriptOperation.Delete):
+                scriptCreateDropOption = 'ScriptDrop';
+                break;
+            case (ScriptOperation.Create):
+                scriptCreateDropOption = 'ScriptCreate';
+            default:
+                scriptCreateDropOption = 'ScriptCreate';
```

Comment:
You can just remove the create option if you're doing to default to ScriptCreate anyways (although realistically I would say you should throw an error instead - there's no good reason to default to create I'd think)